### PR TITLE
Added a new suicide act for pipe valves.

### PR DESCRIPTION
**Pull request by lordpidey**
:cl: LordPidey
   rscadd: Added suicide command for pipe valves.  It's quite messy.
   /:cl:

A video of the process is visible at
https://www.youtube.com/watch?v=gYFvK93DqdA

### CODE REVIEW

File: code/game/machinery/pipe/construction.dm [hunk 1]
```diff
@@ -240,6 +240,21 @@ var/global/list/pipeID2State = list(
 
 	qdel(src)
 
+/obj/item/pipe/suicide_act(mob/user)
+	if (pipe_type in list(PIPE_PUMP, PIPE_PASSIVE_GATE, PIPE_VOLUME_PUMP))
+		user.visible_message("<span class='suicide'>[user] shoved the [src.name] in \he's mouth and turned it on!  It looks like \he's trying to commit suicide.</span>")
```
Comment (neersighted) [inline code/game/machinery/pipe/construction.dm]: No src, just `[name]`

File: code/game/machinery/pipe/construction.dm [hunk 2]
```diff
@@ -240,6 +240,21 @@ var/global/list/pipeID2State = list(
 
 	qdel(src)
 
+/obj/item/pipe/suicide_act(mob/user)
+	if (pipe_type in list(PIPE_PUMP, PIPE_PASSIVE_GATE, PIPE_VOLUME_PUMP))
+		user.visible_message("<span class='suicide'>[user] shoved the [src.name] in \he's mouth and turned it on!  It looks like \he's trying to commit suicide.</span>")
+		if(istype(user, /mob/living/carbon))
+			var/mob/living/carbon/C = user
+			for (var/i = 1 to 20)
+				C.vomit(0,1,0,4,0)
```
Comment (neersighted) [inline code/game/machinery/pipe/construction.dm]: No space in `for(`

File: code/modules/mob/living/carbon/carbon.dm [hunk 3]
```diff
@@ -588,31 +588,44 @@ var/const/GALOSHES_DONT_HELP = 4
...
-		if(blood)
-			if(T)
-				T.add_blood_floor(src)
-			adjustBruteLoss(3)
-		else
-			if(T)
-				T.add_vomit_floor(src)
-			nutrition -= lost_nutrition
-			adjustToxLoss(-3)
+		for (var/i = 0 to distance)
```
Comment (neersighted) [inline code/modules/mob/living/carbon/carbon.dm]: Remove the spaces please.

**Comment by neersighted** [comment]
Looks pretty good. To be clear, is this for any pipe or just for manual valves?

**Comment by lordpidey** [comment]
It will only work for gas pumps, volume pumps, and Passive gates.

A video of the process is visible at
[https://www.youtube.com/watch?v=gYFvK93DqdA](https://www.youtube.com/watch?v=gYFvK93DqdA)

I will go ahead and make the requested changes shortly.

**Comment by MMMiracles** [comment]
brutal af

**Comment by TrustyGun** [comment]
make it spit out organs too for that extra brutal factor

**Comment by Bawhoppen** [comment]
>only does OXYLOSS

did you see how much blood there was do brute too

**Comment by as334** [comment]
Janitors HATE him

File: code/game/machinery/pipe/construction.dm [hunk 4]
```diff
@@ -240,6 +240,21 @@ var/global/list/pipeID2State = list(
 
 	qdel(src)
 
+/obj/item/pipe/suicide_act(mob/user)
+	if (pipe_type in list(PIPE_PUMP, PIPE_PASSIVE_GATE, PIPE_VOLUME_PUMP))
```
Comment (duncathan) [inline code/game/machinery/pipe/construction.dm]: Remind me to redo this

**Comment by duncathan** [comment]
Not bad

File: code/game/machinery/pipe/construction.dm (hunk 4 above)
Comment (lordpidey) [inline code/game/machinery/pipe/construction.dm]: what's wrong with it?

Comment (duncathan) [inline code/game/machinery/pipe/construction.dm]: My code is shitty, not yours

**Pull request merged**